near/nep141-locker/src/lib.rs

```diff
@@ -549,9 +587,94 @@ impl Contract {
 
     fn decode_prover_result(result_idx: u64) -> Result<ProverResult, PromiseError> {
         match env::promise_result(result_idx) {
-            PromiseResult::Successful(data) => Ok(ProverResult::try_from_slice(&data)
-                .unwrap_or_else(|_| env::panic_str("Invalid proof"))),
+            PromiseResult::Successful(data) => {
+                Ok(ProverResult::try_from_slice(&data).sdk_expect("Invalid proof"))
+            }
             PromiseResult::Failed => Err(PromiseError::Failed),
         }
     }
+
+    fn insert_raw_transfer(
+        &mut self,
+        nonce: u128,
+        transfer_message: &TransferMessage,
+        sender_id: &AccountId,
+    ) -> Option<Vec<u8>> {
+        self.pending_transfers.insert_raw(
+            &borsh::to_vec(&nonce).sdk_expect("ERR_BORSH"),
+            &TransferMessageStorage::encode_borsh(transfer_message, sender_id)
+                .sdk_expect("ERR_BORSH"),
+        )
+    }
+
+    fn add_transfer_message(
+        &mut self,
+        nonce: u128,
+        transfer_message: &TransferMessage,
+        message_owner: &AccountId,
+    ) -> NearToken {
+        let storage_usage = env::storage_usage();
+        self.insert_raw_transfer(nonce, transfer_message, message_owner);
+        env::storage_byte_cost().saturating_mul((env::storage_usage() - storage_usage).into())
+    }
+
+    fn remove_transfer_message(&mut self, nonce: u128) -> TransferMessage {
+        let storage_usage = env::storage_usage();
+        let (transfer_message, owner) = self
+            .pending_transfers
+            .remove(&nonce)
+            .map(|m| m.into_main())
+            .sdk_expect("ERR_TRANSFER_NOT_EXIST");
+
+        let refund =
+            env::storage_byte_cost().saturating_mul((storage_usage - env::storage_usage()).into());
```

Comment:
flush?

Reply:
we are using `near-sdk::collections` not `near-sdk::store`